lib/chef/knife/bootstrap.rb

```diff
@@ -588,13 +598,30 @@ def register_client
       def perform_bootstrap(remote_bootstrap_script_path)
         ui.info("Bootstrapping #{ui.color(server_name, :bold)}")
         cmd = bootstrap_command(remote_bootstrap_script_path)
+        bootstrap_run_command(cmd)
+      end
+
+      # Actual bootstrap command to perform on the node.
+      # Handles recursive calls if su USER failed to authenticate.
+      def bootstrap_run_command(cmd)
         r = connection.run_command(cmd) do |data|
           ui.msg("#{ui.color(" [#{connection.hostname}]", :cyan)} #{data}")
         end
         if r.exit_status != 0
-          ui.error("The following error occurred on #{server_name}:")
-          ui.error(r.stderr)
-          exit 1
+          stderr = (r.stderr + r.stdout).strip
+
+          if stderr.match?("su: Authentication failure")
+            ui.warn("Failed to authenticate su - #{config[:su_user]} to #{server_name}")
+            password = ui.ask("Enter password for su - #{config[:su_user]}@#{server_name}:", echo: false)
+
+            set_transport_options(su_password: password)
+
+            bootstrap_run_command(cmd)
+          else
+            ui.error("The following error occurred on #{server_name}:")
+            ui.error(stderr)
+            exit(r.exit_status)
```

Comment:
Instead of `exit 1` I would prefer to add exit with whatever exist_status we have got.